test/box-tap/key_def.test.lua

```diff
@@ -414,9 +415,17 @@ test:test('compare_with_key()', function(test)
 
     -- Unserializable key.
     local exp_err = "unsupported Lua type 'function'"
-    local key = {function() end}
-    local ok, err = pcall(key_def_b.compare_with_key, key_def_b, tuple_a, key)
+    local key_def_c = key_def_lib.new({
+        {type = 'string', fieldno = 1},
+        {type = 'number', fieldno = 2},
+    })
+    local key = {string.rep('abcd', 2^20), 2, function() end}
+    local tuple_c = box.tuple.new({'a', 2, 3})
+    local region_before = fiber.info()[fiber.self().id()].memory.used
+    local ok, err = pcall(key_def_c.compare_with_key, key_def_c, tuple_c, key)
+    local region_after = fiber.info()[fiber.self().id()].memory.used
```

Comment:
Okay, we can test that the fiber region does not leak. However it is the test case for a bugfix, so I'll move it to its own file.

Reply:
I added the separate test case. It looks fragile, though: tarantool side changes can break it. Not sure, whether we should keep it or drop.

I'll re-request review to discuss it with @void234 (which proposes to add the test case) and @Gerold103.

Reply:
It seems, there is a silent agreement to keep the test for the region memory 'leak'. I'll resolve this discussion so.